apps/rig_kafka/lib/rig_kafka/avro.ex

```diff
@@ -0,0 +1,115 @@
+defmodule RigKafka.Avro do
+  @moduledoc """
+  TODO
+  """
+
+  require Logger
+  use Memoize
+  use Rig.Config, [:schema_registry_host]
+
+  @spec parse_schema(String.t()) :: {:ok, map()}
+  def parse_schema(subject) do
+    {:ok, %{"schema" => raw_schema, "id" => id}} = get(subject)
+    schema = :avro.decode_schema(raw_schema)
+    Logger.debug("Using Avro schema with name=#{subject}")
+    {id, schema}
+  end
+
+  @spec parse_schema(integer) :: map()
+  def parse_schema_by_id(id) do
```

Comment:
Yes, `{:ok, map()}` is a mistake. Thing is the second function is returning also `ID` of schema, so if we overload a single function it will require some branching -- or you see some nice way how to have it as a 1 function and avoid if/else here and there?

Or we can use single call by ID, but to me it's more convenient to specify rather name than ID when producing event. What do you think?

Reply:
I've had guards in mind. For example, you could use `is_number` for the ID case and default to string otherwise.